### ISSUE TITLE
light: client stalls reporting block data after a restart

### ISSUE THREAD

**Issue by creachadair** [tendermint/tendermint#7773]
## Summary

Restarting a light client appears to put it into a state where it can no longer correctly respond to queries for valid blocks, if those blocks were not previously queried.

Git bisection shows that this behaviour appears at commit bda948e8, the merge of #6807.

## Repro

The attached [repro.zip](https://github.com/tendermint/tendermint/files/8004498/repro.zip) has a bash script to reproduce these instructions.

In summary:

- Start a node and allow it to generate some blocks (if it doesn't already have any).
- Start a light client at some trusted height against that node.
- Use `curl` to request `/block` info from the light client about valid heights, and observe that it correctly reports the results.
- Stop the light client, and restart it with the same settings.
- Use `curl` to request `/block` info from valid heights, and observe that for heights _previously_ requested, it correctly reports the results, but for _other_ valid heights it fails to report anything, instead logging an error.

**Comment by creachadair** [comment]
Ping @ancazamfir, did I miss any important details about the problem you observed?

**Comment by ancazamfir** [comment]
Looks good! 

> Stop the light client, and restart it with the same settings.


When testing a solution here we should make sure that restarting without any flags also works:
`tendermint light <chain-id>`
As per help: "To restart the node, thereafter only the chainID is required."

**Comment by creachadair** [comment]
> Looks good!
> 
> > Stop the light client, and restart it with the same settings.
> 
> When testing a solution here we should make sure that restarting without any flags also works: `tendermint light <chain-id>` As per help: "To restart the node, thereafter only the chainID is required."

Yes, that's true for _restarting_ the light client, but not for starting it the first time, right?

```
./build/tendermint --home=./tmhome light test-chain-IrF74Y --primary http://localhost:26657 --height=5 --hash=98FCB0303A482743170A7C781C35828A779FA45DA648676AFF9DA40E7A2EEA63 --dir=./tmhome-light
2022-02-07T13:22:12-08:00 INFO Creating client... chainID=test-chain-IrF74Y
Error: no witnesses connected. please reset light client
Usage:
  tendermint light [chainID] [flags]

Examples:
light cosmoshub-3 -p http://52.57.29.196:26657 -w http://public-seed-node.cosmoshub.certus.one:26657
	--height 962118 --hash 28B97BE9F6DE51AC69F70E0B7BFD7E5C9CD1A595B7DC31AFF27C50D4948020CD

```

**Comment by creachadair** [comment]
Also: If you specified an alternate `--dir` for the light client, that will (apparently) also be necessary, in addition to the chain ID. (That's obvious in retrospect, but I found it failed without it in a local test).
